### PR TITLE
add httpsOverHttp to support proxy

### PR DESCRIPTION
**Pull request by yoh1496**
#572 

- adding proxy support ( using `HTTPS_PROXY` environmental value )

I wrote a little code to use this extension under the environment needs to use a proxy.
Code review and consider to support using a proxy please.

### CODE REVIEW

File: src/github/credentials.ts [hunk 1]
```diff
@@ -131,7 +133,14 @@ export class CredentialStore {
 	}
 
 	private createOctokit(type: string, creds: IHostConfiguration): Octokit {
+		const proxySettings = process.env.HTTPS_PROXY ?
+			new URL(process.env.HTTPS_PROXY) : null;
```
Comment (queerviolet) [inline src/github/credentials.ts]: Could we catch the possible parsing error here and `showErrorMessage`?

**Comment by yoh1496** [comment]
@queerviolet Thank you for your code review!

I wrote a try-catch block
and I wrote a code for supporting BASIC-Auth proxy ( I forgot to write in this pull request ).

Code review please, again.

**Comment by queerviolet** [comment]
Hi! This now fails in the case when `HTTPS_PROXY` isn't set. Also, I just realized that we have the exact same agent creation code in two places. We don't really need that—we can create the agent once at startup and forget about it.

I made a PR on your PR with that change: https://github.com/yoh1496/vscode-pull-request-github/pull/1

The diff looks huge, because I also rebased atop the most recent master. Rebase your branch off `Microsoft:master` first, and it'll be a lot less shocking.

**Comment by yoh1496** [comment]
@queerviolet hi, I'm sorry for my late response and dirty commits,,,

I modified a little code given from you and pushed it. Please confirm it.

**Comment by queerviolet** [comment]
Looks good! Congrats :)

**Pull request merged**